File: codes/from_python_to_go/error_handling.py
#!/usr/bin/python3


# let's write a function that handles an exception
def unsafe_division(number1, number2):
    '''Divide number1 by the number2.'''
    return number1 / number2


def main():
    '''Will handles an exception.'''
    try:
        print(unsafe_division(3, 0))
    except ZeroDivisionError as exc:
        print(exc)
    print("Good bye!")

if __name__ == "__main__":
    main()

File: codes/from_python_to_go/error_handling.go
package main

import "fmt"
import "log"

// Unsafe division divide number1 by the number2.
func Unsafe_division(number1, number2 int) (int, error) {
	if number2 == 0 { // HL
		return 0, fmt.Errorf("integer division or modulo by zero") // HL
	} // HL
	return number1 / number2, nil
}

// Main handles an exception.
func main() {
	res, err := Unsafe_division(3, 0)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(res)
	fmt.Println("Good bye!")
}
